Microsoft.DotNet.Interactive.PowerShell/Commands/EnterAzShellCommand.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+
+namespace Microsoft.DotNet.Interactive.PowerShell.Commands
+{
+    /// <summary>
+    /// Connect to Azure PowerShell for code execution.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Enter, "AzShell", DefaultParameterSetName = DefaultSetName)]
+    public sealed class EnterAzShellCommand : PSCmdlet
+    {
+        private const string TenantIdSetName = "TenantIdSpecified";
+        private const string DefaultSetName = "Default";
+
+        [Parameter(ParameterSetName = TenantIdSetName)]
+        public Guid TenantId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultSetName)]
+        public SwitchParameter Reset;
+
+        /// <summary>
+        /// ProcessRecord override.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            var context = KernelInvocationContext.Current;
+            if (context?.CurrentKernel is PowerShellKernel psKernel)
```

Comment:
This API is confusing, I know, but this should probably be `HandlingKernel`, not `CurrentKernel`.

Reply:
Adding comment to `HandlingKernel` and `CurrentKernel` properties should be sufficient.
`CurrentKernel` is the one the notebook running against initially, `HandlingKernel` could be changed when using `#!pwsh` for example, is this understanding correct?